src/avram/query_builder.cr

```diff
@@ -309,6 +309,17 @@ class Avram::QueryBuilder
     self
   end
 
+  def or(&block : Avram::QueryBuilder -> Avram::QueryBuilder)
+    if @wheres.empty? && @raw_wheres.empty?
+      raise Avram::InvalidQueryError.new("Cannot call `or` before calling a `where`")
+    end
+
+    @wheres.last.conjunction = Avram::Where::Conjunction::Or if @wheres.any?
+    @raw_wheres.last.conjunction = Avram::Where::Conjunction::Or if @raw_wheres.any?
+    block.call(self)
```

Comment:
I think there is a bug here that @matthewmcgarvey described. In that there could be doulbe `ors` on accident. For example `MyQuery.new.where("age > 30").name("Bob").or(&.name("Jane"))` it would add the `or` to *both* wheres.

I'm almost thinking we should make `or` ONLY work for `@wheres`. If you want to use an `or` with a raw where, then you can use a raw where to do it `.where("name $1 OR $2")`! What do you think? I think that is less bug-prone and kinda makes sense anyway since the `or` is on queryable anyway

Reply:
I like the idea of making the `or` only work on where and not raw_where. Though, I saw #460 which looks nice and would fix the issue of the ordering. But if we merged that one in, then I'm not sure how that would affect us being able to do different things on `where` vs `raw_where` since that PR would make `raw_where` deprecated 🤔